### PR TITLE
[L0] Fix timestamp event evict after delete

### PR DESCRIPTION
**Pull request by nrspruit**

### CODE REVIEW

**Comment by nrspruit** [comment]
Fix for use after free chance given no event caching.

File: source/adapters/level_zero/event.cpp [hunk 1]
```diff
@@ -1033,6 +1024,16 @@ ur_result_t urEventReleaseInternal(ur_event_handle_t Event) {
     }
   }
 
+  // Save pointer to the queue before deleting/resetting event.
```
Comment (pbalcer) [inline source/adapters/level_zero/event.cpp:1027]: I think this comment was for this line:
`auto Queue = Event->UrQueue;`
:)

**Review by pbalcer — APPROVED** [review]

File: source/adapters/level_zero/event.cpp (hunk 1 above)
Comment (nrspruit) [reply source/adapters/level_zero/event.cpp:1027]: thanks, I will move that comment.

Comment (nrspruit) [reply source/adapters/level_zero/event.cpp]: fixed

**Review by pbalcer — APPROVED** [review]

**Comment by nrspruit** [comment]
timestamps is not in the 0.9.x release so removing that label.

**Review by steffenlarsen — APPROVED** [review]

**Pull request merged**